balancer-design.typ

```diff
@@ -17,3 +17,50 @@ In Rust, packages are called "crates". The Balancer and Harness is split into mu
 	image("figures/balancer-crates.svg"),
 	caption: "Package diagram showing the Balancer and Harness crates.",
 ) <Figure::balancer-crates>
+
+== Dependencies
+
+#let cargo = toml("Cargo.toml")
+#let crates = cargo.workspace.dependencies
+
+#let build_table(data) = {
+  let keys = data.keys()
+	let rows = ()
+
+	for (key, value) in data {
+		if(type(value) == str) {
+			rows.push((key, value, ""))
+		} 
+		else if(type(value) == dictionary) {
+			let row = (key,)
+			
+			if(value.keys().contains("version")) { row.push(value.version) }
+			else if(value.keys().contains("path")) { row.push(value.path) }
+			else if(value.keys().contains("git")) { row.push(value.git) }
+
+			if(value.keys().contains("features")) {
+				let features = value.features.join(", ")
+
+				row.push(features)
+			}
+			else if(value.keys().contains("rev")) { row.push(value.rev) }
```

Comment:
`rev` is part of the version, its not a crate feature. A crate that uses `git` and `rev` to specify the version can also have `features`